miner/miner.go

```diff
@@ -62,8 +64,8 @@ func (miner *Miner) SetEtherbase(addr common.Address) {
 	miner.worker.setEtherbase(addr)
 }
 
-func (miner *Miner) GenerateBlock() (*types.Block, error) {
-	return miner.worker.commitNewWork()
+func (miner *Miner) GenerateBlock(snowCtx *snow.Context, proposerVMBlockCtx *block.Context) (*types.Block, error) {
```

Comment:
I think these two always contexts go together, maybe we should consider creating a new Predicate context and wrap these two.

Reply:
Good call I added `PredicateContext` to package `precompile` so that it can be used in `core` and in `PredicateFunc` in `precompile`.